README.md

```diff
@@ -10,27 +10,81 @@ One of the main objectives of this project is to supply a good starting point fo
 
 <p align="center">
   <img height="250" src="/images/arch.png" >
 </p>
 
-The core libraries for the communication between layer components are: [**LiveData**](https://developer.android.com/topic/libraries/architecture/livedata), used to notify the UI changes and [**Coroutines**](https://kotlinlang.org/docs/reference/coroutines-overview.html), used to perform all background tasks.
+## Overview
+In order to understand how this work together, it's important to talk about each component's role inside the presentation layer.
+
+First of all, the architecture. It's a combination of the MVI and Redux architecture patterns.
+The MVI pattern has three main components: Intent, Model, and View.
+The intent refers to the intention to change the state of the app, so in Gong's case, it would be the actions, delivered then to ViewModel.
+ViewModel holds the model component of the pattern. It is responsible of creation of a new state, which is an immutable data structure.
+At any given moment, there is only one state in the app, which represents a single source of truth.
+The only way to change the state is to create a new one, triggered by the actions. But when and how is the new state created?
+The Redux pattern comes up at this point. Redux is a pattern and library for managing and updating application state, using events called "actions".
+More precisely, the main components of Redux are State, Action, and Reducer.
+In Gong the composables communicate actions to the viewModels so they can manage and emit the state back to view.
+This ensure state can only be updated in a predictable way.
+Then, inside the model, the reducer is called with a proper action and the latest state and forward its result as an output value of the model.
+Reducer is a function that takes the previous state and action and creates a new state, and in Gong this role is played by the function `processAction`, located in viewModels.
+
+
+To continue with the insight of the project, let's see how this is done.
+With the shared flow, actions are broadcast to an unknown number (zero or more) of subscribers.
+In the absence of a subscriber, any posted action is immediately dropped. It is a design pattern to use for actions that must be processed immediately or not at all.
+
+
+The ViewModel handles each action in the `processAction` method. Whenever an action is added to the "contract", it also has to be added here.
+So all actions can be managed from the same place.
+With the channel, each event is delivered to a single subscriber.
+An attempt to post an event without subscribers will suspend as soon as the channel buffer becomes full, waiting for a subscriber to appear. Posted events are never dropped by default.
+Then to handle this Ui effects "oneSHotEvents" are used. Because Channels are hot and it is not necessary to show side effect again when orientation changed or UI become visible again.
```

Comment:
```suggestion
Then to handle this Ui effects "oneShotEvents" are used. Because Channels are hot and it is not necessary to show side effect again when orientation changed or UI become visible again.
```